src/storm-pars/transformer/TimeTravelling.h

```diff
@@ -0,0 +1,95 @@
+                          const boost::optional<std::vector<RationalFunction>>& stateRewardVector, const models::sparse::StateLabeling stateLabelling,
+                          const std::set<std::string> labelsInFormula);
+
+    /**
+     * extendStateLabeling extends the given state labeling to newly created states. It will set the new labels to the labels on the given state.
+     * 
+     * @param oldLabeling The old labeling.
+     * @param oldSize The size of the old labeling.
+     * @param newSize The size of the new labeling (>= oldSize).
+     * @param stateWithLabels The new states will have the labels that this state has.
+     * @param labelsInFormula The labels that occur in the property.
+     * @return models::sparse::StateLabeling 
+     */
+    models::sparse::StateLabeling extendStateLabeling(models::sparse::StateLabeling const& oldLabeling, uint_fast64_t oldSize, uint_fast64_t newSize,
+                                                      uint_fast64_t stateWithLabels, const std::set<std::string> labelsInFormula);
+    /**
+     * Sums duplicate transitions in a vector of MatrixEntries into one MatrixEntry.
+     * 
+     * @param entries 
+     * @return std::vector<storm::storage::MatrixEntry<uint_fast64_t, RationalFunction>> 
+     */
+    std::vector<storm::storage::MatrixEntry<uint_fast64_t, RationalFunction>> joinDuplicateTransitions(
+        std::vector<storm::storage::MatrixEntry<uint_fast64_t, RationalFunction>> const& entries);
+    /**
+     * A preprocessing for time-travelling. It collapses the constant
+     * transitions from a state into a single number that directly goes to the
+     * next parametric transitions.
+     * 
+     * @param state Root for the algorithm.
+     * @param matrix FlexibleMatrix of the pMC.
+     * @param alreadyVisited Recursive argument, set this to the empty map.
+     * @param treeStates The tree states (see updateTreeStates).
+     * @param allParameters The set of all parameters of the pMC.
+     * @param stateRewardVector The state reward vector of the pMC.
+     * @param stateLabelling The state labeling of the pMC.
+     * @param labelsInFormula The labels in the formula.
+     * @return false (returns true in recursive cases)
+     */
+    bool collapseConstantTransitions(uint_fast64_t state, storage::FlexibleSparseMatrix<RationalFunction>& matrix, std::map<uint_fast64_t, bool>& alreadyVisited,
+                    const std::map<RationalFunctionVariable, std::map<uint_fast64_t, std::set<uint_fast64_t>>>& treeStates,
+                    const std::set<carl::Variable>& allParameters, const boost::optional<std::vector<RationalFunction>>& stateRewardVector,
+                    const models::sparse::StateLabeling stateLabelling, const std::set<std::string> labelsInFormula);
+    bool labelsIntersectedEqual(const std::set<std::string>& labels1, const std::set<std::string>& labels2, const std::set<std::string>& intersection);
```

Comment:
I would not make this utility function part of the class, but just define this in the cpp file